state_machine/applications/flight/Tasks/beacon_task.py

```diff
@@ -1,97 +1,40 @@
-            self.debug("If you've attached an antenna, edit '/Tasks/beacon_task.py' to actually beacon", 2)
-            print()  # blank line
-            self.cubesat.radio.listen()
-
-        self.debug("Listening 10s for response (non-blocking)")
-        heard_something = await self.cubesat.radio.await_rx(timeout=10)
-
-        if heard_something:
-            # retrieve response but don't ACK back unless an antenna is attached
-            response = self.cubesat.radio.receive(keep_listening=True, with_ack=ANTENNA_ATTACHED)
-            if response is not None:
-                self.debug("packet received")
-                self.debug('msg: {}, RSSI: {}'.format(response, self.cubesat.radio.last_rssi - 137), 2)
-
-                """
-                ########### ADVANCED ###########
-                Over-the-air commands
-                See beep-sat guide for more details
-                """
-                if len(response) >= 6:
-                    if not ANTENNA_ATTACHED:
-                        self.debug('Antenna not attached. Skipping over-the-air command handling')
-                    else:
-                        if response[:4] == self.super_secret_code:
-                            cmd = bytes(response[4:6])  # [pass-code(4 bytes)] [cmd 2 bytes] [args]
-                            cmd_args = None
-                            if len(response) > 6:
-                                self.debug('command with args', 2)
-                                try:
-                                    cmd_args = response[6:]  # arguments are everything after
-                                    self.debug(f'cmd args: {cmd_args}', 2)
-                                except Exception as e:
-                                    self.debug(f'arg decoding error: {e}', 2)
-                            if cmd in cdh.commands:
-                                try:
-                                    if cmd_args is None:
-                                        self.debug(f'running {cdh.commands[cmd]} (no args)')
-                                        self.cmd_dispatch[cdh.commands[cmd]](self)
-                                    else:
-                                        self.debug(f'running {cdh.commands[cmd]} (with args: {cmd_args})')
-                                        self.cmd_dispatch[cdh.commands[cmd]](self, cmd_args)
-                                except Exception as e:
-                                    self.debug(f'something went wrong: {e}')
-                                    self.cubesat.radio.send(str(e).encode())
-                            else:
-                                self.debug('invalid command!')
-                                self.cubesat.radio.send(b'invalid cmd' + response[4:])
-        else:
-            self.debug('no messages')
-        self.cubesat.radio.sleep()
-        self.debug('finished')
+        tq.push(Message(10, "Hello World!"))
+        if not self.first_time:
```

Comment:
In what case are we meant to use the first_time variable? My understanding is that it checks if this is the first message ever being sent or not. Is this just for us to know if the cubesat has attempted transmission?

Reply:
this is just a testing thing, it only pushes the large message to the tq once.

Reply:
Kinda got the first_time thing backward, fixed in a new commit.